src/node_watchdog.cc

```diff
@@ -90,6 +90,30 @@ void Watchdog::Timer(uv_timer_t* timer) {
   uv_stop(&w->loop_);
 }
 
+void SigintWatchdog::Init(Environment* env, Local<v8::Object> target) {
+  Local<v8::FunctionTemplate> constructor = env->NewFunctionTemplate(New);
+  constructor->InstanceTemplate()->SetInternalFieldCount(1);
+  Local<v8::String> sigintWatchdogString =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "SigintWatchdog");
+    constructor->SetClassName(sigintWatchdogString);
+  target->Set(env->context(), sigintWatchdogString,
+              constructor->GetFunction(env->context()).ToLocalChecked())
+              .Check();
+}
+
+void SigintWatchdog::New(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  auto isolate = args.GetIsolate();
+  auto env = node::Environment::GetCurrent(args);
+  auto self = args.This();
+  auto callback = args[0].As<v8::Function>();
```

Comment:
Please add a `CHECK(args[0]->IsFunction());` before this line. `args[0].As<v8::Function>()` is undefined behavior when `args[0]` isn't actually a function.